src/load.ts

```diff
@@ -392,6 +396,46 @@ export async function lookUpLatestVersion(
   }
 }
 
+export async function lookUpLatestPatchVersion(
+  packageName: string,
+  version: string,
+  log: LogFunction = () => {}
+): Promise<string> {
+  const customRegistry = getCustomRegistry(log);
+  let res: AxiosResponse;
+  try {
+    if (customRegistry) {
+      res = await axiosGet(`${customRegistry.replace(/\/$/, '')}/${packageName}`, {
+        responseType: 'json'
+      });
+    } else {
+      try {
+        res = await axiosGet(`https://packages.fhir.org/${packageName}`, {
+          responseType: 'json'
+        });
+      } catch (e) {
+        // Fallback to trying packages2.fhir.org
+        res = await axiosGet(`https://packages2.fhir.org/packages/${packageName}`, {
+          responseType: 'json'
+        });
+      }
+    }
+  } catch {
+    throw new LatestVersionUnavailableError(packageName, customRegistry, true);
+  }
+
+  if (res?.data?.versions) {
+    const versions = Object.keys(res.data.versions);
+    const latest = maxSatisfying(versions, version);
```

Comment:
Should we do one more check to ensure that the version is in the right format (i.e., `/\d+\.\d+\.x/`)? As it is now, if we passed in a version like `2.x`, it would give us the latest minor version. Since the function specifically says latest patch version, and since we export it, I think it makes sense to do the check so we're not false advertising.

Reply:
Ohh yes, we definitely should.

Reply:
Okay I looked at this briefly and I feel like this could actually go two ways:
- I could do what you suggested and add the check for the version format at the top of the function to ensure only patch wild cards are passed in.
- Alternatively, I could rename this function to be `lookupLatestWildcardVersion` and not include that check. There is a check in `mergeDependency` that would ensure that implementation only gets wildcard versions for patch wildcards. Part of me likes this approach because now that we're using semver in the function, there's no reason it needs to be specific to patch versions anymore. However, FHIR only allows patch wildcards, so I also understand wanting to restrict the functionality of this function.

I could go either way, so if you have an opinion, let me know.

Reply:
Yeah, I had thought about that as well, and honestly, I'm also fine with either way. If we went w/ option 2, maybe I'd name it `lookupMaxSatisfyingVersion` or `resolveSemverVersion` -- as it works for any semver (not just wildcards).

Reply:
@cmoesel  - Today I'm leaning toward the first option, just because the second option makes it seem like we'd potentially support wildcard functionality that we don't have any intention of supporting (because FHIR doesn't support it). Plus, if we did have that function support any semver, maybe we should add tests to ensure that too, and that seems unnecessary for our use case. The first option seems to show the intended functionality more easily.

I'll go with the first option, unless you disagree.

Reply:
@jafeltra - Works for me!